src/clj/puppetlabs/services/jruby/jruby_puppet_schemas.clj

```diff
@@ -108,7 +115,8 @@
    :borrow-timeout schema/Int
    :max-active-instances schema/Int
    :max-requests-per-instance schema/Int
-   :use-legacy-auth-conf schema/Bool})
+   :use-legacy-auth-conf schema/Bool
+   (schema/optional-key :environment-class-cache-enabled) schema/Bool})
```

Comment:
I don't know that we have anything about the config yet, I think some of that will probably just come out in the wash as we're doing the initial repo setup.  Creating a ticket and throwing it into that epic seems fine.

As far as this PR goes, dissoc seems OK and ye should feel free to merge when it's ready.